samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemProducer.scala

```diff
@@ -21,134 +21,100 @@ package org.apache.samza.system.kafka
 
 
 import java.util.concurrent.atomic.AtomicReference
-import java.util.concurrent.{TimeUnit, ConcurrentHashMap, Future}
+import java.util.concurrent.TimeUnit
 
 import org.apache.kafka.clients.producer.Callback
 import org.apache.kafka.clients.producer.Producer
 import org.apache.kafka.clients.producer.ProducerRecord
 import org.apache.kafka.clients.producer.RecordMetadata
 import org.apache.kafka.common.PartitionInfo
-import org.apache.samza.SamzaException
+import org.apache.kafka.common.errors.SerializationException
 import org.apache.samza.system.OutgoingMessageEnvelope
 import org.apache.samza.system.SystemProducer
+import org.apache.samza.system.SystemProducerException
 import org.apache.samza.util.ExponentialSleepStrategy
 import org.apache.samza.util.KafkaUtil
 import org.apache.samza.util.Logging
 import org.apache.samza.util.TimerUtils
 
-import scala.collection.JavaConverters._
+object KafkaSystemProducer {
+  // Send messages
+  val SendMsgFormat = "Enqueuing message: %s, %s."
+  val StreamNameNullOrEmptyErrorMsg = "Stream Name should be specified in the stream configuration file."
+  val PreviousErrorMsg = "Producer was unable to recover from previous exception."
+  val NullProducerMsg = "Kafka producer is null."
+  val FailedSendMsgFormat = "Failed to send message for Source: %s on System:%s Topic:%s Partition:%s"
+  val DropErrorMsg = "Ignoring producer exception. All messages in the failed producer request will be dropped!"
+  val NewProducerMsgFormat = "Creating a new producer for system %s."
+
+  // Flush messages
+  val FailedFlushMsg = "Flush failed. One or more batches of messages were not sent!"
+
+  // Stop messages
+  val StopMsg = "Stopping producer for system: "
+  val UnhandledSendErrorMsg = "Observed an earlier send() error while closing producer"
```

Comment:
Minor: Maybe s/earlier/unhandled?

Reply:
That wouldn't always be accurate, since the exception may have been thrown to the sender many times (and swallowed by user code) before finally propagating up to the container from flush(). In that case, we're just logging the error (which the application has already seen and ignored) on shutdown for troubleshooting purposes.